### PR TITLE
[MOD-8039] Reorganize stats files

### PR DESCRIPTION
**Pull request by meiravgri**
This PR reorganizes the files related to the module info to improve structure and readability.

# Elaborate `info` dir
## Introduce `info_redis.*` files
these files are responsible to aggregate and output all the shard info to `INFO MODULES` command.
* Move `RS_moduleInfoFunc` implementation from `module-init.c`
This function is registered via `RedisModuleAPI` to be called upon `INFO MODULES` (or `INFO EVERYTHING`)
* Implement each info section in a static function to enhance readability
* Relocate `*_AddToInfo` 
  * Moved from global_stats.c
  `FieldsGlobalStats_AddToInfo`
`TotalGlobalStats_Queries_AddToInfo`
`DialectsGlobalStats_AddToInfo`
`IndexesGlobalStats_AddToInfo`
  * Moved from config.c:
`RSConfig_AddToInfo`

## Introduce` indexes_info.*` files
Files dedicated to aggregating shard indexes information.
* Move structs from `info_command.h` to `indexes_info.*`
* Rename structs: 
  * `TotalSpecsInfo` -> `TotalIndexesInfo`
  * `TotalSpecsFieldInfo` -> `TotalIndexesFieldsInfo`
* Exposes `IndexesInfo_TotalInfo`. The implementation was moved from `redisearch_api.c::RediSearch_TotalInfo`
* `redisearch_api.c::RediSearch_TotalInfo` now calls` index_info::IndexesInfo_TotalInfo 

## global_stats.* files
* move to `info/` dir
* Relocat all `*_AddToInfo` functions:
Functions responsible for writing stats to INFO MODULES have been moved to INFO_MODULES.c.
* introduce `QueriesGlobalStats` to hold queries information
* introduce `TotalGlobalStats_GetQueryStats` 
**Atomically** reads and copies query stats into a QueriesGlobalStats struct, which is returned to the caller.

## info_command.* files
* moved to info dir 
* move structs to `indexes_info`

# Additional changes
## util/units.h
Contains units macro definitions
* Move `CLOCKS_PER_MILLISEC` from `profile.c` and `nfo_command.h` to `util/units.h`
* `MEMORY_MB`

## new test
added a unit test for `redisearch_API::RediSearch_TotalInfo`

### CODE REVIEW

File: src/config.c [hunk 1]
```diff
@@ -1099,31 +1099,6 @@ int RSConfig_SetOption(RSConfig *config, RSConfigOptions *options, const char *n
   return rc;
 }
 
-void RSConfig_AddToInfo(RedisModuleInfoCtx *ctx) {
```
Comment (meiravgri) [inline src/config.c:1102]: mvoed to info/INFO_MODULES.c

File: src/info/indexes_info.c [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * Copyright Redis Ltd. 2016 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "indexes_info.h"
+#include "util/dict.h"
+#include "spec.h"
+
+TotalIndexesInfo IndexesInfo_TotalInfo() {
```
Comment (meiravgri) [inline src/info/indexes_info.c:11]: moved from redisearch_api

File: src/info/indexes_info.h [hunk 3]
```diff
@@ -43,10 +42,11 @@ typedef struct TotalSpecsInfo {
   size_t num_active_indexes_indexing;  // Number of active write indexes
   size_t total_active_writes;          // Total number of active writes
   size_t total_active_queries;         // Total number of active queries (reads)
-} TotalSpecsInfo;
+} TotalIndexesInfo;
```
Comment (meiravgri) [inline src/info/indexes_info.h:45]: both struct were moved here from info_command

File: src/info/info_command.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+ */
+
+#include "redismodule.h"
+#ifndef INFO_COMMAND_H
+#define INFO_COMMAND_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
```
Comment (meiravgri) [inline src/info/info_command.h:14]: moved structs to indexes_info

File: src/module-init/module-init.c [hunk 5]
```diff
@@ -86,90 +86,6 @@ static int initAsLibrary(RedisModuleCtx *ctx) {
   return REDISMODULE_OK;
 }
 
-#define MEMORY_HUMAN(x) ((x) / (double)(1024 * 1024))
```
Comment (meiravgri) [inline src/module-init/module-init.c:89]: moved macro definition to util/units

File: src/module-init/module-init.c [hunk 6]
```diff
@@ -86,90 +86,6 @@ static int initAsLibrary(RedisModuleCtx *ctx) {
...
-
-void RS_moduleInfoFunc(RedisModuleInfoCtx *ctx, int for_crash_report) {
```
Comment (meiravgri) [inline src/module-init/module-init.c:91]: moved to INFO_MODULES

File: src/profile.h [hunk 7]
```diff
@@ -9,8 +9,6 @@
 #include "value.h"
 #include "aggregate/aggregate.h"
 
-#define CLOCKS_PER_MILLISEC (CLOCKS_PER_SEC / 1000)
```
Comment (meiravgri) [inline src/profile.h:12]: moved to "util/units.h"

File: src/redisearch_api.c [hunk 8]
```diff
@@ -913,59 +914,8 @@ size_t RediSearch_MemUsage(RSIndex* rm) {
 }
 
 // Collect statistics of all the currently existing indexes
-TotalSpecsInfo RediSearch_TotalInfo(void) {
```
Comment (meiravgri) [inline src/redisearch_api.c:916]: moved to info/indexes_info

File: src/info/global_stats.h [hunk 9]
```diff
@@ -64,24 +68,20 @@ void FieldsGlobalStats_UpdateStats(FieldSpec *fs, int toAdd);
...
- * Add all the query-related information to the INFO command.
- */
-void TotalGlobalStats_Queries_AddToInfo(RedisModuleInfoCtx *ctx);
-
-/**
- * Add all the dialect-related information to the INFO command.
+ * Safely reads and returns a copy of the global queries stats.
  */
-void DialectsGlobalStats_AddToInfo(RedisModuleInfoCtx *ctx);
+QueriesGlobalStats TotalGlobalStats_GetQueryStats();
```
Comment (meiravgri) [inline src/info/global_stats.h:84]: atomically reads queries stats

File: src/util/units.h [hunk 10]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright Redis Ltd. 2016 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+
+#pragma once
+
+#define CLOCKS_PER_MILLISEC (CLOCKS_PER_SEC / 1000)
+#define MEMORY_HUMAN(x) ((x) / (double)(1024 * 1024))
```
Comment (GuyAv46) [inline src/util/units.h:10]: Can we drop it? or rename to `MEMORY_MB`?

Comment (meiravgri) [reply src/util/units.h:10]: No, it's in use.
I'll rename

**Review by GuyAv46 — APPROVED** [review]

**Pull request merged**